### PR TITLE
verification data

### PR DESCRIPTION
**Pull request by TheDeveloper101**
added functions to generate new verification data and random strings for verification

### CODE REVIEW

File: squire_sdk/src/server/accounts.rs [hunk 1]
```diff
@@ -107,3 +110,18 @@ where
     state.load_user(user);
     StatusCode::ACCEPTED
 }
+
+pub async fn new_verification_data(key: String, user: User) -> VerificationData {
```
Comment (TylerBloom) [inline squire_sdk/src/server/accounts.rs]: There is no need for this to be `async`, and it should probably be a method called `new` for `VerificationData`. It also doesn't need to take a user. Associating a user with some verification data is the job of the server state, not the data itself. Please remove the user argument.

Comment (TheDeveloper101) [reply squire_sdk/src/server/accounts.rs]: alright, and yes i independently removed the user argument in my latest commit

File: squire_sdk/src/server/accounts.rs [hunk 2]
```diff
@@ -107,3 +110,26 @@ where
...
+impl VerificationData {
+    fn new(key: String) -> VerificationData{
+        VerificationData {
+            confirmation: key.to_owned(),
+            status: false,
+        }
+    }
+}
+
+pub async fn generate_key() -> String {
```
Comment (TylerBloom) [inline squire_sdk/src/server/accounts.rs:123]: This does not need to be `async`.

File: squire_sdk/src/server/accounts.rs [hunk 3]
```diff
@@ -107,3 +110,26 @@ where
...
+    thread_rng()
+        .sample_iter(&Alphanumeric)
+        .take(6)
+        .map(char::from)
+        .collect()
+}
+
+pub async fn attempt_verification(account: SquireAccount) -> VerificationRequest {
```
Comment (TylerBloom) [inline squire_sdk/src/server/accounts.rs:131]: This does not need to be `async`.

**Review by TylerBloom — COMMENTED** [review]
After these two changes, this should be good to merge.

**Review by TylerBloom — APPROVED** [review]

**Pull request merged**